lib/ipam/ipam_block_reader_writer.go

```diff
@@ -120,9 +107,14 @@ func (rw blockReaderWriter) findUnclaimedBlock(ctx context.Context, host string,
 		for subnet := blocks(); subnet != nil; subnet = blocks() {
 			// Check if a block already exists for this subnet.
 			log.Debugf("Getting block: %s", subnet.String())
-			if _, ok := exists[subnet.String()]; !ok {
-				log.Infof("Found free block: %+v", *subnet)
-				return subnet, nil
+			_, err := rw.queryBlock(ctx, *subnet, "")
```

Comment:
Related to merge conflict above, we don't want to query the block on every iteration here.